### DISCUSSION TITLE
1.7-pre1

### DISCUSSION THREAD

**Discussion by DreamVoid** [DreamVoid/MiraiMC#168] [Polls]
这是一个**预发布**版本，部分功能仍在测试，请及时反馈在本版本发现的问题。

**此版本对事件的更改极大，完全弃用了原有的事件，虽然插件仍然对旧事件提供支持，但这个支持并不完美。如果插件使用遇到问题，请不要升级并等待其他依赖MiraiMC的插件的开发者更新。**

# 更新日志
### 新增
* 添加 ```getMembers()``` 获取群成员列表方法
* **分类所有事件到新的包**
* MessagePreSendEvent 事件现在支持取消
* 增加云控调整插件工作方式

### 优化和修复
* 优化绑定功能，解决错误使用 String 存储 UUID 的情况
* 移出 PlaceholderAPI 模块为独立项目
* 优化Mirai核心加载流程
* 代码细节优化

### 优化和修复
* 增加额外的Before方法
* 修正代码语法错误

### 优化和修复
* 修复停用插件时重复关闭两次机器人的问题

### 优化和修复
* 增加服务器关闭的插件处理

由Github生成的版本说明见下

---

## What's Changed
* 优化绑定功能 by @DreamVoid in https://github.com/DreamVoid/MiraiMC/pull/157
* 分类所有事件到新的包 by @DreamVoid in https://github.com/DreamVoid/MiraiMC/pull/158
* 增加云控调整插件工作方式 by @DreamVoid in https://github.com/DreamVoid/MiraiMC/pull/167

**Full Changelog**: https://github.com/DreamVoid/MiraiMC/compare/v1.6.1...v1.7-pre1

---

# 下载说明
* ```MiraiMC-Bukkit.jar``` - 适用于CraftBukkit（如Spigot、Paper）服务端
* ```MiraiMC-Bungee.jar``` - 适用于BungeeCord（如WaterFall）服务端
* ```MiraiMC-Nukkit.jar``` - 适用于NukkitX服务端
* ```MiraiMC-Sponge.jar``` - 适用于SpongeVanilla和SpongeForge服务端
* ```MiraiMC-Velocity.jar``` - 适用于Velocity服务端

<hr /><em>This discussion was created from the release <a href='https://github.com/DreamVoid/MiraiMC/releases/tag/v1.7-pre1'>1.7-pre1</a>.</em>

**Comment by devsong42** [comment]
`Exception in unnamed coroutine.
java.lang.StackOverflowError: null
	at me.dreamvoid.miraimc.bukkit.event.MiraiBotOnlineEvent.<init>(MiraiBotOnlineEvent.java:13) ~[MiraiMC-Bukkit.jar:?]
	at me.dreamvoid.miraimc.bukkit.event.bot.MiraiBotOnlineEvent.<init>(MiraiBotOnlineEvent.java:21) ~[MiraiMC-Bukkit.jar:?]
	at`
疯狂报错，后面一直重复13行和21行的异常

**Reply by devsong42** [reply]
这是刚开服的，然后每处理一条mirai sendgroupmessage的信息又疯狂报错
`An exception occurred when processing event. Subscriber scope: '<unnamed>'. Broadcaster scope: '<unnamed>'
java.lang.StackOverflowError: null
	at me.dreamvoid.miraimc.bukkit.event.MiraiGroupMessagePreSendEvent.<init>(MiraiGroupMessagePreSendEvent.java:13) ~[MiraiMC-Bukkit.jar:?]
	at me.dreamvoid.miraimc.bukkit.event.message.presend.MiraiGroupMessagePreSendEvent.<init>(MiraiGroupMessagePreSendEvent.java:20) ~[MiraiMC-Bukkit.jar:?]`
后面也是一样的重复这两个异常

**Reply by devsong42** [reply]
qq群到服务器的信息处理目前不知道，还没人发信息

**Comment by devsong42** [comment]
还发现在加载的时候会花费好几分钟，不知道怎么回事
`[05:01:21] [Server thread/INFO]: [MiraiMC] Loading MiraiMC v1.7-pre1
[05:05:41] [Server thread/INFO]: [MiraiMC] Verifying mirai-core-all-2.11.0.jar`

**Reply by DreamVoid** [reply]
> <img alt="" width="32" height="32" src="https://avatars.githubusercontent.com/u/63984647?s=64&amp;v=4">
> devsong42

更新最新开发版，速度问题缓解了
